src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/OffenderSearchService.kt

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+import org.springframework.web.server.ResponseStatusException
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto.offendersearch.OffenderSearchPrisoner
+
+@Service
+class OffenderSearchService(
+  private val offenderSearchWebClient: WebClient,
+  private val offenderSearchClientCredentials: WebClient,
+  private val objectMapper: ObjectMapper,
+) {
+
+  private fun getClient(useClientCredentials: Boolean = false): WebClient {
+    return if (useClientCredentials) offenderSearchClientCredentials else offenderSearchWebClient
+  }
+
+  /**
+   * Search prisoners by their prisoner number
+   *
+   * Requires ROLE_GLOBAL_SEARCH or ROLE_PRISONER_SEARCH role.
+   */
+  fun searchByPrisonerNumbers(
```

Comment:
So would use client creds to call this - the client will be given the right roles

Reply:
https://github.com/ministryofjustice/hmpps-non-associations-api/pull/35/commits/2d6f8b6a6752a1008daad07a4d47555c5df1f2ff